ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl.hs

```diff
@@ -142,14 +145,14 @@ openDBInternal args launchBgTasks = do
         immBlockNo = ChainSel.getImmBlockNo secParam chain immDbTipBlockNo
 
     atomically $ LgrDB.setCurrent lgrDB ledger
-    varChain          <- atomically $ newTVar chain
-    varImmBlockNo     <- atomically $ newTVar immBlockNo
-    varIterators      <- atomically $ newTVar Map.empty
-    varReaders        <- atomically $ newTVar Map.empty
-    varNextIteratorId <- atomically $ newTVar $ IteratorId 0
-    varNextReaderId   <- atomically $ newTVar 0
-    varCopyLock       <- atomically $ newTMVar ()
-    varBgThreads      <- atomically $ newTVar []
+    varChain          <- newTVarWithInvariantM  isNF chain
+    varImmBlockNo     <- newTVarWithInvariantM  isNF immBlockNo
+    varIterators      <- newTVarM Map.empty
+    varReaders        <- newTVarM Map.empty
+    varNextIteratorId <- newTVarWithInvariantM  isNF (IteratorId 0)
+    varNextReaderId   <- newTVarWithInvariantM  isNF 0
+    varCopyLock       <- newTMVarWithInvariantM isNF ()
+    varBgThreads      <- newTVarWithInvariantM  isNF []
```

Comment:
These fail as some things are simply not in normal form.

Reply:
Let's revert those invariants, merge this PR, and then reintroduce the invariants and figure out where that is going wrong.